docs/src/modules/components/Link.js

```diff
@@ -8,17 +8,17 @@ import MuiLink from '@material-ui/core/Link';
 import { useUserLanguage } from 'docs/src/modules/utils/i18n';
 
 const NextComposed = React.forwardRef(function NextComposed(props, ref) {
-  const { as, href, ...other } = props;
+  const { linkAs, href, ...other } = props;
```

Comment:
We had conflict here with the `as` prop from emotion.

Reply:
I think we should document the props that are now intercepted by emotion. It is a breaking change if you were passing these through. Maybe add a section in the changelog that only lists affected components and once we migrated all we just give the general advise that if you had `<MuiComponent component={SomeOtherComponent} emotionProp1 emotionProp2 />`, that these are no longer passed to `SomeOtherComponent`

Reply:
Will we need to update the Next.js and Gatsby examples for the conflict?

Reply:
@eps1lon will add an entry in the `migration-v4.md` for it. @oliviertassinari good call, let me check if the examples need to be updated.

Reply:
Added migration entry - https://github.com/mui-org/material-ui/pull/23841/commits/d764665b9e7020ac2e02762bb7ffe76fcfc76482 we can extend it with more props if there are any. At this moment this is the only one I could notice not being propagated.

Both `nextjs` examples have been updated. I didn't noticed anything worth updating in the gatsby examples.